packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -0,0 +1,144 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {Result} from '../../api/search/search/result';
+import {isArray, removeDuplicates} from '../../utils/utils';
+import {executeSearch, fetchMoreResults} from '../search/search-actions';
+import {registerFolding} from './folding-actions';
+import {
+  FoldedResult,
+  FoldingFields,
+  getFoldingInitialState,
+} from './folding-state';
+
+export interface ResultWithFolding extends Result {
+  parentResult: ResultWithFolding | null;
+  childResults: ResultWithFolding[];
+}
+
+function getRawValue(result: ResultWithFolding, field: string) {
+  const rawValue = result.raw[field] as string | [string] | undefined;
+  if (isArray(rawValue)) {
+    return rawValue[0];
+  }
+  return rawValue;
+}
+
+function getParentField(result: ResultWithFolding, fields: FoldingFields) {
+  return getRawValue(result, fields.parent);
+}
```

Comment:
Having dedicated functions is a step in the right direction, but I also think we should remove `getRawValue` so that it makes it clearer to future developers which field is holding an array, and which is holding a string.